ExampleMod/Content/Tiles/Plants/ExampleCactus.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Tiles
+{
+	public class ExampleCactus : ModCactus
+	{
+		public override void SetStaticDefaults() {
+			// Makes Example Cactus grow on ExampleOre
+			GrowsOnTileId = new int[1] { ModContent.TileType<ExampleOre>() };
+		}
+
+		public override Asset<Texture2D> GetTexture() => ModContent.Request<Texture2D>("ExampleMod/Content/Tiles/Plants/ExampleCactus", AssetRequestMode.ImmediateLoad);
```

Comment:
Reminder to replace immediate load with same code used in another PR so that modders don't need to spec it